diff --git a/node_modules/ol/geom/flat/geodesic.js b/node_modules/ol/geom/flat/geodesic.js
index c567143..e9b63bb 100644
--- a/node_modules/ol/geom/flat/geodesic.js
+++ b/node_modules/ol/geom/flat/geodesic.js
@@ -142,10 +142,11 @@ export function greatCircleArc(
  * @param {number} lat2 Latitude 2.
  * @param {import("../../proj/Projection.js").default} projection Projection.
  * @param {number} squaredTolerance Squared tolerance.
+ * @param {import("../../proj/Projection.js").default|undefined} sourceProjection Source projection.
  * @return {Array<number>} Flat coordinates.
  */
-export function meridian(lon, lat1, lat2, projection, squaredTolerance) {
-  const epsg4326Projection = getProjection('EPSG:4326');
+export function meridian(lon, lat1, lat2, projection, squaredTolerance, sourceProjection) {
+  sourceProjection = sourceProjection || getProjection('EPSG:4326');
   return line(
     /**
      * @param {number} frac Fraction.
@@ -154,7 +155,7 @@ export function meridian(lon, lat1, lat2, projection, squaredTolerance) {
     function (frac) {
       return [lon, lat1 + (lat2 - lat1) * frac];
     },
-    getTransform(epsg4326Projection, projection),
+    getTransform(sourceProjection, projection),
     squaredTolerance,
   );
 }
@@ -166,10 +167,11 @@ export function meridian(lon, lat1, lat2, projection, squaredTolerance) {
  * @param {number} lon2 Longitude 2.
  * @param {import("../../proj/Projection.js").default} projection Projection.
  * @param {number} squaredTolerance Squared tolerance.
+ * @param {import("../../proj/Projection.js").default|undefined} sourceProjection Source projection.
  * @return {Array<number>} Flat coordinates.
  */
-export function parallel(lat, lon1, lon2, projection, squaredTolerance) {
-  const epsg4326Projection = getProjection('EPSG:4326');
+export function parallel(lat, lon1, lon2, projection, squaredTolerance, sourceProjection) {
+  sourceProjection = sourceProjection || getProjection('EPSG:4326');
   return line(
     /**
      * @param {number} frac Fraction.
@@ -178,7 +180,7 @@ export function parallel(lat, lon1, lon2, projection, squaredTolerance) {
     function (frac) {
       return [lon1 + (lon2 - lon1) * frac, lat];
     },
-    getTransform(epsg4326Projection, projection),
+    getTransform(sourceProjection, projection),
     squaredTolerance,
   );
 }
diff --git a/node_modules/ol/layer/Graticule.js b/node_modules/ol/layer/Graticule.js
index 6291ed0..2fd448a 100644
--- a/node_modules/ol/layer/Graticule.js
+++ b/node_modules/ol/layer/Graticule.js
@@ -169,6 +169,7 @@ const INTERVALS = [
  * [30, 10]
  * ```
  * @property {boolean} [wrapX=true] Whether to repeat the graticule horizontally.
+ * @property {import("../proj/Projection.js").default|string|undefined} [gridProjection] The Projection used to draw the graticule lines.
  * @property {Object<string, *>} [properties] Arbitrary observable properties. Can be accessed with `#get()` and `#set()`.
  */
 
@@ -209,6 +210,7 @@ class Graticule extends VectorLayer {
     delete baseOptions.lonLabelStyle;
     delete baseOptions.latLabelStyle;
     delete baseOptions.intervals;
+    delete baseOptions.gridProjection;
     super(baseOptions);
 
     /**
@@ -217,6 +219,17 @@ class Graticule extends VectorLayer {
      */
     this.projection_ = null;
 
+    /**
+     * @type {import("../proj/Projection.js").default}
+     * @private
+     */
+    this.gridProjection_ =
+      options.gridProjection === undefined
+        ? getProjection('EPSG:4326')
+        : typeof options.gridProjection === 'string'
+        ? getProjection(options.gridProjection)
+        : options.gridProjection;
+
     /**
      * @type {number}
      * @private
@@ -1019,9 +1032,13 @@ class Graticule extends VectorLayer {
     /** @type {Array<number>} **/
     const p2 = [];
     for (let i = 0, ii = this.intervals_.length; i < ii; ++i) {
-      const delta = clamp(this.intervals_[i] / 2, 0, 90);
+      const delta = clamp(this.intervals_[i] / 2, 0, this.maxLat_);
       // Don't attempt to transform latitudes beyond the poles!
-      const clampedLat = clamp(centerLat, -90 + delta, 90 - delta);
+      const clampedLat = clamp(
+        centerLat,
+        this.minLat_ + delta,
+        this.maxLat_ - delta
+      );
       p1[0] = centerLon - delta;
       p1[1] = clampedLat - delta;
       p2[0] = centerLon + delta;
@@ -1053,6 +1070,7 @@ class Graticule extends VectorLayer {
       maxLat,
       this.projection_,
       squaredTolerance,
+      this.gridProjection_,
     );
     let lineString = this.meridians_[index];
     if (!lineString) {
@@ -1123,6 +1141,7 @@ class Graticule extends VectorLayer {
       maxLon,
       this.projection_,
       squaredTolerance,
+      this.gridProjection_,
     );
     let lineString = this.parallels_[index];
     if (!lineString) {
@@ -1181,9 +1200,7 @@ class Graticule extends VectorLayer {
    * @private
    */
   updateProjectionInfo_(projection) {
-    const epsg4326Projection = getProjection('EPSG:4326');
-
-    const worldExtent = projection.getWorldExtent();
+    const worldExtent = this.gridProjection_.getExtent();
 
     this.maxLat_ = worldExtent[3];
     this.maxLon_ = worldExtent[2];
@@ -1193,7 +1210,7 @@ class Graticule extends VectorLayer {
     // If the world extent crosses the dateline define a custom transform to
     // return longitudes which wrap the dateline
 
-    const toLonLatTransform = getTransform(projection, epsg4326Projection);
+    const toLonLatTransform = getTransform(projection, this.gridProjection_);
     if (this.minLon_ < this.maxLon_) {
       this.toLonLatTransform_ = toLonLatTransform;
     } else {
@@ -1218,7 +1235,7 @@ class Graticule extends VectorLayer {
     // Transform the extent to get the limits of the view projection extent
     // which should be available to the graticule
 
-    this.fromLonLatTransform_ = getTransform(epsg4326Projection, projection);
+    this.fromLonLatTransform_ = getTransform(this.gridProjection_, projection);
     const worldExtentP = applyTransform(
       [this.minLon_, this.minLat_, this.maxLon_, this.maxLat_],
       this.fromLonLatTransform_,
@@ -1243,9 +1260,8 @@ class Graticule extends VectorLayer {
     // Some projections may have a void area at the poles
     // so replace any NaN latitudes with the min or max value closest to a pole
 
-    this.projectionCenterLonLat_ = this.toLonLatTransform_(
-      getCenter(projection.getExtent()),
-    );
+    this.projectionCenterLonLat_ = getCenter(this.gridProjection_.getExtent());
+
     if (isNaN(this.projectionCenterLonLat_[1])) {
       this.projectionCenterLonLat_[1] =
         Math.abs(this.maxLat_) >= Math.abs(this.minLat_)
